fix(blocks): handle enter key on image

remidej opened the pull request:
<!--
Hello 👋 Thank you for submitting a pull request.

To help us merge your PR, make sure to follow the instructions below:

- Create or update the tests
- Create or update the documentation at https://github.com/strapi/documentation
- Refer to the issue you are closing in the PR description: Fix #issue
- Specify if the PR is ready to be merged or work in progress (by opening a draft PR)

Please ensure you read the Contributing Guide: https://github.com/strapi/strapi/blob/main/CONTRIBUTING.md
-->

### What does it do?

Inserts an empty paragraph when pressing enter while having the selection on an image

### Why is it needed?

Right now, pressing enter on an image deletes it. Reported on discord

### How to test it?

Create image, click it, press enter, new

Boegie19 commented:
@remidej seems almost the same as https://github.com/strapi/strapi/issues/18655

remidej commented:
@Boegie19 yes I think the combination of this and #18840 should fix it, but I'll check manually once both are merged to be sure

madhurisandbhor commented on packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Image.tsx:
```diff
@@ -162,6 +161,12 @@ const imageBlocks: Pick<BlocksStore, 'image'> = {
     matchNode: (node) => node.type === 'image',
     isInBlocksSelector: true,
+    handleEnterKey(editor) {
+      Transforms.insertNodes(editor, {
+        type: 'paragraph',
+        children: [{ type: 'text', text: '' }],
+      });
```
if we are adding empty block on enter, then we should remove empty block added on creation, otherwise it adds 2 empty blocks of no use.

madhurisandbhor commented:
Do you think [this](https://strapi-inc.atlassian.net/browse/CS-403) could be added in this PR if its a small change

simotae14 reviewed (commented):
If you add more than 1 asset the empty line after the images is not created

remidej commented:
@madhurisandbhor I added the focus style thing, it wasn't too long

@simotae14 I just updated the PR to remove the inserted empty paragraph altogether, as recommended by Madhuri. Now users shouldn't need it with the enter key fix
